### ISSUE TITLE
Certain file-based app restore errors do not surface in the editor

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#81611]
Repro:

```cs
#:package Newtonsoft.Json@13.0.3;

using Newtonsoft.Json;

Console.WriteLine("Hello World!");
```

Note the `;` at the end of the directive. In this case the package does not restore and no errors are reported visibly in the editor.

Relevant bit of the logs:

```log
2025-12-09 15:16:40.313 [info] [workspace/didChangeWatchedFiles] [LanguageServerProjectLoader] Successfully completed load of f:\src\fbp-bug-bash\app4.cs
2025-12-09 15:16:40.315 [info] [workspace/didChangeWatchedFiles] [LanguageServerProjectLoader] Restoring app4.cs: Running dotnet restore on f:\src\fbp-bug-bash\app4.cs
2025-12-09 15:16:41.050 [info] [workspace/didChangeWatchedFiles] [LanguageServerProjectLoader] Restoring app4.cs:   Determining projects to restore...
2025-12-09 15:16:41.209 [info] [workspace/didChangeWatchedFiles] [LanguageServerProjectLoader] Restoring app4.cs: C:\Program Files\dotnet\sdk\10.0.100\NuGet.targets(196,5): error : '13.0.3;' is not a valid version string.
2025-12-09 15:16:41.264 [info] [workspace/didChangeWatchedFiles] [LanguageServerProjectLoader] Restoring app4.cs: Failed to run restore on f:\src\fbp-bug-bash\app4.cs
2025-12-09 15:16:41.266 [info] [workspace/didChangeWatchedFiles] [LanguageServerProjectLoader] Restore complete
2025-12-09 15:16:41.266 [info] [workspace/didChangeWatchedFiles] [LanguageServerProjectLoader] Completed (re)load of all projects in 00:00:02.6877098
2025-12-09 15:16:42.015 [info] [workspace/didChangeWatchedFiles] [Microsoft.CodeAnalysis.MSBuild.BuildHostProcessManager] .NET BuildHost started from c:\Users\rigibson\.vscode\extensions\ms-dotnettools.csharp-2.103.33-win32-x64\.roslyn\Microsoft.CodeAnalysis.LanguageServer.exe reloading to start from C:\Program Files\dotnet\dotnet.exe to match necessary SDK location.
2025-12-09 15:16:43.095 [info] [workspace/didChangeWatchedFiles] [LanguageServerProjectLoader] Successfully completed load of f:\src\fbp-bug-bash\app4.cs
2025-12-09 15:16:43.095 [info] [workspace/didChangeWatchedFiles] [LanguageServerProjectLoader] Completed (re)load of all projects in 00:00:01.7227109
```
